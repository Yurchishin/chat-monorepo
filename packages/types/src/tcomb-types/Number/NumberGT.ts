import t from 'tcomb-validation'
import R from '@chat/ramda'

export const NumberGT = (min: number) => t.refinement(
    t.Number,
    R.gt(R.__, min),
    'NumberGT',
)
