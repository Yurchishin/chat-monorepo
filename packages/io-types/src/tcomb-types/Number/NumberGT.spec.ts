import { NumberGT } from './NumberGT'
import { expectTypeSuccess, expectTypeFailure } from "../../tests"

describe('NumberGT(5)', () => {
	const GT5 = NumberGT(5)

	expectTypeSuccess(GT5, 6000)
	expectTypeFailure(GT5, 5)
	expectTypeFailure(GT5, 4)
	expectTypeFailure(GT5, '1')
})
