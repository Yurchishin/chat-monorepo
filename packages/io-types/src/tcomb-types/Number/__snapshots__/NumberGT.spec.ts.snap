// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberGT(5) expectTypeFailure(1): TYPE_FAILURE::NumberGT(1) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "1",
      "expected": [Function],
      "message": "Invalid value \\"1\\" supplied to Number",
      "path": Array [],
    },
  ],
  "value": "1",
}
`;

exports[`NumberGT(5) expectTypeFailure(4): TYPE_FAILURE::NumberGT(4) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": 4,
      "expected": [Function],
      "message": "Invalid value 4 supplied to NumberGT",
      "path": Array [],
    },
  ],
  "value": 4,
}
`;

exports[`NumberGT(5) expectTypeFailure(5): TYPE_FAILURE::NumberGT(5) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": 5,
      "expected": [Function],
      "message": "Invalid value 5 supplied to NumberGT",
      "path": Array [],
    },
  ],
  "value": 5,
}
`;
